feat/IT-17-FE-RegisrationEmailVerification

siuLongBao opened the pull request:
feat: IT-17-Regisration Email Verification (Front End)

Implemented front-end pages for registration email verification and resend verification link on desktop and mobile.
- Loading page
![Picture3](https://github.com/user-attachments/assets/f66cea3e-f724-4aac-b23f-505075fa3913)

- If successful, will show verification success and then directly go to login page.
![Picture1](https://github.com/user-attachments/assets/ce517f86-d61f-408b-9bf0-adff99ae6c60)
![Picture2](https://github.com/user-attachments/assets/bfc6cc8d-609c-46ab-a803-0a39e7a26dd4)

- If fail (wrong token), will show verification fail and then allow user to either request for resend verification email or go to login page
![image](https://github.com/user-attachments/assets/35b876fe-08c0-45c2-9200-863e6d00a156)

- If network error, will show server error and then allow user to either request for resend verification email or go to login page
![Picture4](https://github.com/user-attachments/assets/09ca32d7-162d-4076-8ad7-47a91fbff75c)

- Resend Verification Email UI and successful implementation of resend verification email
![image](https://github.com/user-attachments/assets/3cb6957d-699c-479d-8875-6fbdfebdc6da)

Feature Implemented:
- Created the email verification page and resend verification email functionality. 
- The email verification page includes loading, success, and failure states. 
- Created a resend verification email functionality allows users to request a new verification email if they did not receive the first one. 
- Both pages have been tested and are fully functional.

siuLongBao commented:
Verifying Page Photo:
![Verifying](https://github.com/user-attachments/assets/b8f39089-4232-4221-8ffb-072074768d59)

Server Error:
![ServerError](https://github.com/user-attachments/assets/0a0e6b3f-1ac7-458a-9f05-c4d2b16578ee)

Verification Success:
![Verification Success](https://github.com/user-attachments/assets/e390c1af-52a9-4874-a9e8-1728a08c48c5)

StevenRen929 commented:
Please follow the Zeplin to do your UI, The Error Message and the selection pop-up should be 'grouped' by a container(In Chakra Ui we use Box). The box should have the boarder and all of those components  should not be like what you show in "Verification Success:" Image.  Please resolve the conflict between your branch and dev branch before you push this commit again. Thanks.

siuLongBao commented:
> Please follow the Zeplin to do your UI, The Error Message and the selection pop-up should be 'grouped' by a container(In Chakra Ui we use Box). The box should have the boarder and all of those components should not be like what you show in "Verification Success:" Image. Please resolve the conflict between your branch and dev branch before you push this commit again. Thanks.

Done!

siuLongBao committed:
Feat: IT-17: Created Resend and Verify Email Page
- Created the email verification page and resend verification email functionality.
    - Email verification page includes loading, success, and failure states.
- Created a resend verification email function
    - Let user request a new verification email if first one does not work.
- All pages have been tested and are fully functional.

siuLongBao commented:
The email verification has been changed as requested.

1. Verification Fail (Desktop):
![Web Verify Fail](https://github.com/user-attachments/assets/2a7f3c82-ad8e-49a7-9f47-769413a7b036)

2. Loading (Desktop):
![Web Loading](https://github.com/user-attachments/assets/20886e31-1431-48a8-8339-188b00145607)

3. Verification Success (Desktop):
![Web Verify Success](https://github.com/user-attachments/assets/ff423bf0-e857-480c-a0a4-55571d4730ac)

4. Verification Fail (Mobile):
![Verify Fail Mobile](https://github.com/user-attachments/assets/92919229-13e5-484d-b107-f2cdc4795d0a)

5. Verifying (Mobile):
![Verifying Mobile](https://github.com/user-attachments/assets/31e3742d-dc9f-4c0f-9392-d41497d22d1d)

6. Verification Success (Mobile):
![Verify Success Mobile](https://github.com/user-attachments/assets/53021734-0483-4b68-bd4d-2def885a91ef)
